### ISSUE TITLE
7$: Error on 'for (var {x} = init in xs)'

### ISSUE THREAD

**Issue by strager** [quick-lint/quick-lint-js#622]
```js
for (var x = init in xs) {}    // OK
for (var {x} in xs) {}         // OK
for (var {x} = init in xs) {}  // BAD
```
The BAD example is illegal because initializers are only allowed if the initializer is preceded by a BindingIdentifier: https://262.ecma-international.org/11.0/#sec-initializers-in-forin-statement-heads
